assets/src/edit-story/components/movable/index.js

```diff
@@ -0,0 +1,141 @@
+
+function Movable( {
+	rotationAngle,
+	x,
+	y,
+	type,
+	targetEl,
+	pushEvent,
+	setPushEvent,
+} ) {
+	const moveable = useRef();
+
+	const {
+		actions: { setPropertiesOnSelectedElements },
+	} = useStory();
+
+	useEffect( () => {
+		if ( moveable.current ) {
+			// If we have persistent event then let's use that.
+			if ( pushEvent ) {
+				moveable.current.moveable.dragStart( pushEvent );
+				// Remove the event right after using, not to influence other elements.
+				setPushEvent( null );
+			}
+			moveable.current.updateRect();
+		}
+		// Disable reason: we should not run this when pushEvent changes!
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [ targetEl, moveable ] );
+
+	const frame = {
+		translate: [ 0, 0 ],
+		rotate: rotationAngle,
+	};
+
+	const setStyle = ( target ) => {
+		target.style.transform = `translate(${ frame.translate[ 0 ] }px, ${ frame.translate[ 1 ] }px) rotate(${ frame.rotate }deg)`;
+	};
+
+	/**
+	 * Resets Movable once the action is done, sets the initial values.
+	 *
+	 * @param {Object} target Target element.
+	 */
+	const resetMoveable = ( target ) => {
+		frame.translate = [ 0, 0 ];
+		setStyle( target );
+		if ( moveable.current ) {
+			moveable.current.updateRect();
+		}
+	};
+
+	return (
+		<Moveable
+			ref={ moveable }
+			target={ targetEl }
+			draggable={ true }
+			resizable={ true }
+			rotatable={ true }
+			onDrag={ ( { target, beforeTranslate } ) => {
+				frame.translate = beforeTranslate;
+				setStyle( target );
+			} }
+			throttleDrag={ 0 }
+			onDragStart={ ( { set } ) => {
+				set( frame.translate );
+			} }
+			onDragEnd={ ( { target } ) => {
+				// When dragging finishes, set the new properties based on the original + what moved meanwhile.
+				const newProps = { x: x + frame.translate[ 0 ], y: y + frame.translate[ 1 ] };
+				setPropertiesOnSelectedElements( newProps );
+				resetMoveable( target );
+			} }
+			onResizeStart={ ( { setOrigin, dragStart } ) => {
+				setOrigin( [ '%', '%' ] );
+				if ( dragStart ) {
+					dragStart.set( frame.translate );
+				}
+			} }
+			onResize={ ( { target, width, height, drag } ) => {
+				target.style.width = `${ width }px`;
+				target.style.height = `${ height }px`;
+				frame.translate = drag.beforeTranslate;
+				setStyle( target );
+			} }
+			onResizeEnd={ ( { target } ) => {
+				setPropertiesOnSelectedElements( {
+					width: parseInt( target.style.width ),
+					height: parseInt( target.style.height ),
+					x: x + frame.translate[ 0 ],
+					y: y + frame.translate[ 1 ],
+				} );
+				resetMoveable( target );
+			} }
+			onRotateStart={ ( { set } ) => {
+				set( frame.rotate );
+			} }
+			onRotate={ ( { target, beforeRotate } ) => {
+				frame.rotate = beforeRotate;
+				setStyle( target );
+			} }
+			onRotateEnd={ () => {
+				setPropertiesOnSelectedElements( { rotationAngle: frame.rotate } );
+			} }
+			origin={ false }
+			pinchable={ true }
+			keepRatio={ 'image' === type }
```

Comment:
The more I think about this, the more I'm convinced we should just have `keepRatio` property in our element's model. Because even images do not have to keep ratio when the "lock" checkmark unset in the edit panel.

Reply:
Makes sense to me 👍

Reply:
Perhaps for now (within this PR), we can keep it as it is? Since currently we don't have the lock mark and without that, it makes more sense for the images always to keep ratio.

Reply:
👍